moped-editor/src/views/projects/projectView/ProjectComponents/EditAttributesModal.js

```diff
@@ -171,16 +144,21 @@ const EditAttributesModal = ({
     editDispatch({ type: "cancel_attributes_edit" });
   };
 
-  const initialFormValues = {
-    component: componentToEdit,
-    subcomponents: componentToEdit?.moped_proj_components_subcomponents,
-    description: componentToEdit?.description,
-    phase: componentToEdit?.moped_phase,
-    subphase: componentToEdit?.moped_subphase,
-    completionDate: componentToEdit?.completion_date,
-    tags: componentToEdit?.moped_proj_component_tags,
-    srtsId: componentToEdit?.srts_id,
-  };
+  const initialFormValues = componentToEdit
+    ? {
+        component: makeComponentFormFieldValue(componentToEdit),
+        description: componentToEdit.description,
+        subcomponents: makeSubcomponentsFormFieldValues(
+          componentToEdit.moped_proj_components_subcomponents
+        ),
+        signal: makeSignalFormFieldValue(componentToEdit),
+        phase: makePhaseFormFieldValue(componentToEdit.moped_phase),
+        subphase: makeSubphaseFormFieldValue(componentToEdit.moped_subphase),
+        completionDate: componentToEdit.completion_date,
+        srtsId: componentToEdit.srts_id,
+        tags: makeTagFormFieldValues(componentToEdit.moped_proj_component_tags),
```

Comment:
These helpers are all extracted from the previous `useInitialValuesOnAttributesEdit` custom hook removed later in this diff. Now, `EditAttributesModal` is in charge of making the shape of data needed for the form. The form is no longer responsible for doing all of this work.